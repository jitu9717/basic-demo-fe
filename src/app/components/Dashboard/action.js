import ACTION from '../../../constants/actions';

// Action Creater to Update list of containers
export const getBeerList = (beerList) => {
    return {
        type: ACTION.GET_BEER_LIST,
        payload: beerList,
    }
};

