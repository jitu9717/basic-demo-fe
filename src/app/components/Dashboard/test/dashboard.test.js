import React from 'react';
import { fetchBeerList } from '../async.action';
import { shallow } from 'enzyme';
import {DashboardContainer, mapStateToProps, mapDispatchToProps} from '../index';
import LoadingIndicator from '../../Common/LoadingIndicator';
import BeerList from '../../BeerList';


jest.mock('../async.action');

const FETCH_ACTION = Symbol('');
fetchBeerList.mockReturnValue(FETCH_ACTION);


describe('Dashboard component should render correctly', () => {
	let allProps,
		dashboardContainer;

	beforeEach(() => {
		allProps = {
			beerList: [{beerType : "Aler Beer", id: 1}],
			fetchBeerList: () => new Promise((resolve, reject)=>{}),
		};
		dashboardContainer = shallow(<DashboardContainer {...allProps}  />);
	});

	it('renders correctly without beerList', () => {
		allProps.beerList = [];
		expect(dashboardContainer).toMatchSnapshot();
	});

    describe("rendered `LoadingIndicator Component`", () => {
        it("always renders a `LoadingIndicator Component`", () => {
            expect(dashboardContainer.find(LoadingIndicator).length).toBe(1);
        });

        it("LoadingIndicator receive 1 props", () => {
            const loader = dashboardContainer.find(LoadingIndicator);
            expect(Object.keys(loader.props()).length).toBe(1);
        });
    });

    describe("rendered `BeerList Component`", () => {
        it("always renders a `BeerList Component`", () => {
            expect(dashboardContainer.find(BeerList).length).toBe(1);
        });

        it("BeerList receive 1 props", () => {
            const beerlist = dashboardContainer.find(BeerList);
            expect(Object.keys(beerlist.props()).length).toBe(1);
        });
    });



});

describe('mapStateToProps ', () => {
	it('should render correct object', () => {
		expect(mapStateToProps).toBeDefined();
	});
});

describe('mapDispatchToProps', () => {
	const dispatch = jest.fn().mockImplementation(x => x);
	const actionDispatchers = mapDispatchToProps(dispatch);

	it('should render the correct list of methods', () => {
	   expect(mapDispatchToProps()).toEqual({
		   fetchBeerList: expect.any(Function),
	   })
	});

	it('returns the result of dispatching fetchBeerList correctly', () => {
		expect(actionDispatchers.fetchBeerList()).toBe(FETCH_ACTION);
		expect(dispatch).toHaveBeenCalledWith(FETCH_ACTION);
	});
});





