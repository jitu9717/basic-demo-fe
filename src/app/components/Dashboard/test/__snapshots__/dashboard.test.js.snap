// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component should render correctly renders correctly without beerList 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dashboard
    beerList={
        Array [
            Object {
              "beerType": "Aler Beer",
              "id": 1,
            },
          ]
    }
    fetchBeerList={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <LoadingIndicator
          showLoader={true}
/>,
        <BeerList
          beerList={
                    Array [
                              Object {
                                "beerType": "Aler Beer",
                                "id": 1,
                              },
                            ]
          }
/>,
      ],
      "className": "pageLayout",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "showLoader": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "beerList": Array [
            Object {
              "beerType": "Aler Beer",
              "id": 1,
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <LoadingIndicator
            showLoader={true}
/>,
          <BeerList
            beerList={
                        Array [
                                    Object {
                                      "beerType": "Aler Beer",
                                      "id": 1,
                                    },
                                  ]
            }
/>,
        ],
        "className": "pageLayout",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "showLoader": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "beerList": Array [
              Object {
                "beerType": "Aler Beer",
                "id": 1,
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
